File: frontend/types/payment.ts
export type PaymentMethod = "CARD" | "CASH_ON_DELIVERY" | "WALLET";
export type PaymentStatus = "PENDING" | "SUCCESS" | "FAILED" | "REFUNDED";

export type Payment = {
  paymentId: string;
  orderId: string;
  userId: string;
  amount: number;
  paymentMethod: PaymentMethod;
  paymentStatus: PaymentStatus;
  transactionReference: string;
  notes: string | null;
  createdAt: string;
  paidAt: string | null;
};

export type CreatePaymentRequest = {
  orderId: string;
  paymentMethod: PaymentMethod;
  notes?: string;
};

File: Payment Service/Payment-Service/src/main/java/com/example/Payment/Service/entities/Payment.java
package com.example.Payment.Service.entities;

import com.example.Payment.Service.enums.PaymentMethod;
import com.example.Payment.Service.enums.PaymentStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "payment_record")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Payment {

    @Id
    @GeneratedValue(generator = "UUID")
    @UuidGenerator
    private UUID id;

    @Column(nullable = false, unique = true)
    private UUID orderId;

    @Column(nullable = false)
    private UUID userId;

    @Column(nullable = false)
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentMethod paymentMethod;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus paymentStatus;

    @Column(nullable = false, unique = true)
    private String transactionReference;

    private String notes;

    private LocalDateTime createdAt;

    private LocalDateTime paidAt;
}
